### PR TITLE
Updated Upstream (Bukkit/CraftBukkit)

### PR DESCRIPTION
**Pull request by Machine-Maker**
```[tasklist]
### TODO
- [x] Remove/Deprecate TrimPattern/TrimMaterial Translatable
- [x] AbstractArrow#getItem vs AbstractArrow#getItemStack
- [x] LivingEntity#getActiveItem vs LivingEntity#getItemInUse
```

----

Upstream has released updates that appear to apply and compile correctly. This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
f02baa38 PR-988: Add World#getIntersectingChunks(BoundingBox)
9321d665 Move getItemInUse up to LivingEntity
819eef73 PR-959: Add access to current item's remaining ticks
c4fdadb0 SPIGOT-7601: Add AbstractArrow#getItem
be8261ca Add support for Java 22
26119676 PR-979: Add more translation keys
66753362 PR-985: Correct book maximum pages and characters per page documentation
c8be92fa PR-980: Improve getArmorContents() documentation
f1120ee2 PR-983: Expose riptide velocity to PlayerRiptideEvent

CraftBukkit Changes:
dfaa89bbe PR-1369: Add World#getIntersectingChunks(BoundingBox)
51bbab2b9 Move getItemInUse up to LivingEntity
668e09602 PR-1331: Add access to current item's remaining ticks
a639406d1 SPIGOT-7601: Add AbstractArrow#getItem
0398930fc SPIGOT-7602: Allow opening in-world horse and related inventories
ffd15611c SPIGOT-7608: Allow empty lists to morph to any PDT list
2188dcfa9 Add support for Java 22
45d6a609f SPIGOT-7604: Revert "SPIGOT-7365: DamageCause blocked by shield should trigger invulnerableTime"
06d915943 SPIGOT-7365: DamageCause blocked by shield should trigger invulnerableTime
ca3bc3707 PR-1361: Add more translation keys
366c3ca80 SPIGOT-7600: EntityChangeBlockEvent is not fired for frog eggs
06d0f9ba8 SPIGOT-7593: Fix sapling growth physics / client-side updates
45c2608e4 PR-1366: Expose riptide velocity to PlayerRiptideEvent
29b6bb79b SPIGOT-7587: Remove fixes for now-resolved MC-142590 and MC-109346

### CODE REVIEW

**Review by kennytv — APPROVED** [review]

**Comment by kennytv** [comment]
(for the current changes, no amending so it's easier to review plz)

File: patches/api/0117-LivingEntity-Active-Item-API.patch [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -202,15 +202,19 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      *
+      * @return the item being used by the player, or null if they are not using
+      * an item
++     * @deprecated Use {@link #getActiveItem()}
+      */
+     @Nullable
++    @Deprecated // Paper
```
Comment (kennytv) [inline patches/api/0117-LivingEntity-Active-Item-API.patch]: For removal on all of these

File: patches/api/0117-LivingEntity-Active-Item-API.patch [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
++
++    /**
++     * Get how long the {@link #getActiveItem()} has been in use for.
++     *
++     * @return time used in ticks
++     */
++    int getActiveItemUsedTime();
++
++    /**
++     * Get the hande using the active item. Will be either
```
Comment (Lulu13022002) [inline patches/api/0117-LivingEntity-Active-Item-API.patch]: Typo: hande -> hand

File: patches/api/0117-LivingEntity-Active-Item-API.patch [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
++     * {@link org.bukkit.inventory.EquipmentSlot#HAND} or
++     * {@link org.bukkit.inventory.EquipmentSlot#OFF_HAND}.
++     *
++     * @return the hand being used
++     */
++    org.bukkit.inventory.@NotNull EquipmentSlot getActiveItemHand();
++
++    /**
++     * Gets remaining time a player needs to keep hands raised with an item to finish using it.
++     * @return Remaining ticks to use the item
```
Comment (Lulu13022002) [inline patches/api/0117-LivingEntity-Active-Item-API.patch]: Missing a newline between description/return

File: patches/api/0117-LivingEntity-Active-Item-API.patch [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
++    @Deprecated(forRemoval = true, since = "1.20.4") // Paper
+     public void setItemInUseTicks(int ticks);
+ 
+     /**
+@@ -850,4 +856,100 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      */
+     void setShieldBlockingDelay(int delay);
+     // Paper end
++
++    // Paper start - active item API
```
Comment (Lulu13022002) [inline patches/api/0117-LivingEntity-Active-Item-API.patch:64]: Letter case are not consistent across all descriptions of return/param in the methods of this block

File: patches/server/0328-Entity-Jump-API.patch [hunk 5]
```diff
@@ -50,15 +50,15 @@ index aba20a4352d8983b01ab5d329187588f68d3e405..aac60e85cd6dba7d87f4a1663c2c6295
...
 +++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
-@@ -935,5 +935,19 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
-     public org.bukkit.inventory.EquipmentSlot getHandRaised() {
-         return getHandle().getUsedItemHand() == net.minecraft.world.InteractionHand.MAIN_HAND ? org.bukkit.inventory.EquipmentSlot.HAND : org.bukkit.inventory.EquipmentSlot.OFF_HAND;
+@@ -961,4 +961,19 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         return org.bukkit.craftbukkit.CraftEquipmentSlot.getHand(this.getHandle().getUsedItemHand());
      }
+     // Paper end - active item API
 +
-+    @Override
```
Comment (Lulu13022002) [inline patches/server/0328-Entity-Jump-API.patch:61]: Why the override is gone here?

**Review by lynxplay — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Pull request merged**